### PR TITLE
added register Boolean to listen

### PR DESCRIPTION
**Pull request by Hotkeyyy**

### CODE REVIEW

**Review by bluefireoly — CHANGES_REQUESTED** [review]
you forgot to add this new parameter to the documentation of the listen function

File: src/main/kotlin/net/axay/kspigot/event/KSpigotListeners.kt [hunk 1]
```diff
@@ -61,16 +61,18 @@ inline fun <reified T : Event> SingleListener<T>.register(
  * @param T the type of event to listen to
  * @param priority the priority when multiple listeners handle this event
  * @param ignoreCancelled if manual cancellation should be ignored
+ * @param register if the event should be registered
```
Comment (bluefireoly) [inline src/main/kotlin/net/axay/kspigot/event/KSpigotListeners.kt:64]: if the event should be registered immediately*

**Review by bluefireoly — CHANGES_REQUESTED** [review]

File: src/main/kotlin/net/axay/kspigot/event/KSpigotListeners.kt (hunk 1 above)
Comment (Hotkeyyy) [reply src/main/kotlin/net/axay/kspigot/event/KSpigotListeners.kt:64]: digga

File: src/main/kotlin/net/axay/kspigot/event/KSpigotListeners.kt [hunk 2]
```diff
@@ -61,16 +61,18 @@ inline fun <reified T : Event> SingleListener<T>.register(
...
     priority: EventPriority = EventPriority.NORMAL,
     ignoreCancelled: Boolean = false,
+    register: Boolean = true,
     crossinline onEvent: (event: T) -> Unit
 ): SingleListener<T> {
     val listener = object : SingleListener<T> {
         override fun onEvent(event: T) = onEvent.invoke(event)
     }
-    listener.register(priority, ignoreCancelled)
+    if(register) listener.register(priority, ignoreCancelled)
```
Comment (bluefireoly) [inline src/main/kotlin/net/axay/kspigot/event/KSpigotListeners.kt:76]: add a space after the `if` keyword LOL

**Review by bluefireoly — CHANGES_REQUESTED** [review]

File: src/main/kotlin/net/axay/kspigot/event/KSpigotListeners.kt (hunk 2 above)
Comment (Hotkeyyy) [reply src/main/kotlin/net/axay/kspigot/event/KSpigotListeners.kt:76]: digga
wtf

**Pull request merged**